koku/api/apps.py

```diff
@@ -14,8 +14,70 @@
 #    You should have received a copy of the GNU Affero General Public License
 #    along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
+"""API application configuration module"""
+
+import logging
+
 from django.apps import AppConfig
 
+from koku.env import ENVIRONMENT
+
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
 
 class ApiConfig(AppConfig):
+    """API application configuration"""
+
     name = 'api'
+
+    def ready(self):
+        """Action on application startup."""
+        # noqa: E402 pylint: disable=C0413
+        from django.db.utils import OperationalError
+        try:
+            self.startup_status()
+            self.check_and_create_service_admin()
+        except OperationalError as op_error:
+            logger.error('Error: %s.', op_error)
+
+    def startup_status(self):  # pylint: disable=R0201
+        """Log the status of the server at startup."""
+        # noqa: E402 pylint: disable=C0413
+        from api.status.model import Status
+        status_info = None
+        status_count = Status.objects.count()
+        if status_count == 0:
+            status_info = Status.objects.create()
+            status_info.save()
+        else:
+            status_info = Status.objects.get(pk=1)
+
+        status_info.startup()
+
+    def create_service_admin(self, service_email):  # pylint: disable=R0201
+        """Create the Service Admin."""
+        # noqa: E402 pylint: disable=C0413
+        from django.contrib.auth.models import User
+        service_user = ENVIRONMENT.get_value('SERVICE_ADMIN_USER',
+                                             default='admin')
+        service_pass = ENVIRONMENT.get_value('SERVICE_ADMIN_PASSWORD',
+                                             default='pass')
+
+        User.objects.create_superuser(service_user,
+                                      service_email,
+                                      service_pass)
+        logger.info('Created Service Admin: %s.', service_email)
```

Comment:
I assume this will change when we start using token auth, but do we have a plan for securely extracting the token out of this auto-generating superuser step?

Reply:
You can get access to the secrets created in OSD. That should give us the user/password to call the token endpoint. We could write a tool that does this.